### ISSUE TITLE
`Building: aarch64-pc-windows-msvc` got `section count exceeds max` warning

### ISSUE THREAD

**Issue by Wyvern** [rust-lang/rust#150045]
Rust: nightly
build with `-Zbuild-std` for `aarch64-pc-windows-msvc` got this warning like below:
note: requested on the command line with `-W linker-messages`

```
Building: aarch64-pc-windows-msvc
warning: linker stdout: C:\a\Img\Img\build-dir\aarch64-pc-windows-msvc\src\build\Img\8c598e69be55a61b\deps\Img.exe : warning LNK4072: section count 104 exceeds max (96); image may not run
```
but `arm64ec-pc-windows-msvc` has no warning, so weird.
I'm sure it's harmless or not.

**Comment by Wyvern** [comment]
`Are you using #[link_section] anywhere?`
NO

`too many naked functions`
there's no naked fn in my project.

**Comment by Fulgen301** [comment]
Can you provide the source code / a reproducer that emits the warning for you?
